app/controllers/admin/merchant_items_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Admin::MerchantItemsController < Admin::BaseController
+  def index
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    merchant = Merchant.find(params[:merchant_id])
+    item = Item.find(params[:item_id])
+    if item.active? == true
+      item.update(active?:false)
+      flash[:notice] = "#{item.name} is no longer for sale"
+    else
+      item.update(active?:true)
+      flash[:notice] = "#{item.name} is now available for sale"
+    end
+    redirect_to "/admin/merchants/#{merchant.id}/items"
```

Comment:
a good candidate for a refactor or pulling this logic out, this is a great method!